### DISCUSSION TITLE
lab4-1 synthesis problem

### DISCUSSION THREAD

**Discussion by sungsungwu** [bol-edu/HLS-SOC-Discussions#126] [Q&A]
不好意思，我想請問一下，請問此次lab該如何做合成呢?

請問是要建立一個新的vivado專案，再將counter.v以及bram.v下去合成嗎?
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/cd6eb145-64a8-4be4-bd35-1545e32c8c56)

以下是我們跑合成時發生的問題，我們有將vivado資料夾內的三個.v檔與bram.v下去合成，counter.v改為我們所寫的專案，會發生以下錯誤，請問該如何解決呢?
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/73c92101-5e0c-4389-8c9b-f3d082653713)

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/f31835b4-4268-4bcc-996a-32b19add5474)

感謝!

**Comment by Lanxas1018** [comment]
同學好，只要將改寫自己設計的counter.v跟bram.v一起合成就好，不是user folder下的檔案都要一起合成

**Comment by sungsungwu** [comment]
你好，不好意思我們只將counter.v跟bram.v合成也會發生以下問題，想請問這樣該怎麼辦，如需提供counter.v的source code也可以，感激不盡!
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/76789e0d-7751-41d8-ad01-94663a9fdedf)

**Reply by Lanxas1018** [reply]
不曉得你有沒有注意到`MPRJ_IO_PADS這個parameter以及default_nettype none這兩個地方，首先`MPRJ_IO_PADS這個parameter是定義在rtl/header/defines.v下方，但我們只將counter.v和bram.v拿到vivado做合成，這個參數自然會被判定成undefined，所以需要在counter.v上方自己定義；non-net type的error是因為default_nettype none造成的，只需將none更改成wire即可。

**Reply by sungsungwu** [reply]
好的，剛剛依照指示合成成功了，感謝您詳細的教學與幫助!
